plugins/vli/fu-vli-pd-common.c

```diff
@@ -20,15 +20,19 @@ fu_vli_pd_common_guess_device_kind(guint32 fwver)
 		return FU_VLI_DEVICE_KIND_VL101;
 	if (tmp == 0x07 || tmp == 0x08)
 		return FU_VLI_DEVICE_KIND_VL102;
-	if (tmp == 0x09)
+	if (tmp == 0x09 || tmp == 0x0a)
 		return FU_VLI_DEVICE_KIND_VL103;
 	if (tmp == 0x0b)
 		return FU_VLI_DEVICE_KIND_VL104;
-	if (tmp == 0xa0)
+	if (tmp == 0x0c)
 		return FU_VLI_DEVICE_KIND_VL105;
-	if (tmp == 0xb0)
+	if (tmp == 0x0d)
 		return FU_VLI_DEVICE_KIND_VL106;
-	if (tmp == 0xc0)
+	if (tmp == 0x0e)
 		return FU_VLI_DEVICE_KIND_VL107;
+	if (tmp == 0xa1 || tmp == 0xb1)
+		return FU_VLI_DEVICE_KIND_VL108;
+	if (tmp == 0xa2 || tmp == 0xb2)
+		return FU_VLI_DEVICE_KIND_VL109;
```

Comment:
sooo how does this work? the top nibble is set, and `guint32 tmp = (fwver & 0x0f000000) >> 24` only looks at the lower half...

Reply:
I've pushed a fix along with some self tests. Does that work?

Reply:
Sorry for the inconvenience.
We use lower half (0~e) for VL100~VL107, but no more value for VL108, VL109...
We change to use whole byte for VL108, VL109...
So the code will be:  
fu_vli_pd_common_guess_device_kind(guint32 fwver)  {
guint32 tmp = (fwver & 0xff000000) >> 24;   // for VL108/109... case
if ( tmp < 0xA0 ) tmp = tmp & 0x0f; // for VL100~VL107 cases
...
}

Reply:
Does my naive code work too?

Reply:
no only "guint32 tmp = (fwver & 0xFF000000) >> 24;" is  not enough,
you must have "if ( tmp < 0xA0 ) tmp = tmp & 0x0f;" too. 
emily will push correction later today.